tx.go

```diff
@@ -558,7 +558,9 @@ func (tx *Tx) writeMeta() error {
 	lg := tx.db.Logger()
 	buf := make([]byte, tx.db.pageSize)
 	p := tx.db.pageInBuffer(buf, 0)
+	tx.db.metalock.Lock()
```

Comment:
It isn't correct. 

We don't need acquire metalock here, it's updating tx's local meta page.

Reply:
+1, i also thought the same, but u know better 👍🏽

Reply:
iiuc bbolt uses CoW, which means the updates will be written to a new pages, and only upon `commit` the metadata list will be updated to point to the new pages

@ahrtr know better these details 👍🏽

Reply:
> But then how about the `writeAt` below?

Only one write TXN is allowed at a time.



> Or that's the question of how `pwrite()` updates the page. Not sure about that.

pls refer to https://github.com/ahrtr/etcd-issues/blob/master/docs/cncf_storage_tag_etcd.md#storage-boltdb-feature

Reply:
Yes, it's a valid point, although it's highly unlikely in practice. We have two meta pages, usually Readonly TXN won't read the same meta page as the RW TXN at the same time; even it does, if it reads some dirty data or partially written data, then the checksum won't match, so it still fallback to the other meta page. It's exactly the reason why we never see any issue in the [concurrent test](https://github.com/etcd-io/bbolt/blob/main/concurrent_test.go)

https://github.com/etcd-io/bbolt/blob/092ee98898c0386b73f520e065238181cdba383b/db.go#L1128-L1140

Reply:
> it's highly unlikely in practice

I agree with that, the amount of data is small, there is validation, yet it's still a race that can happen. Threads can be stalled at any point on a busy machine as well in which case chances can get higher. Like consider validation done successfully in `meta()`, but then the reader thread paused and waiting in the queue while writers change the contents before or in the middle of reader doing a `Copy()`.